idea-plugin/src/main/kotlin/arrow/meta/plugin/idea/plugins/higherkinds/HigherKindsIdePlugin.kt

```diff
@@ -20,32 +30,28 @@ val IdeMetaPlugin.higherKindsIdePlugin: Plugin
     meta(
       addLineMarkerProvider(
         icon = ArrowIcons.HKT,
-        matchOn = { it.safeAs<KtClass>()?.takeIf(::isHigherKindedType) },
-        message = { classOrInterface: KtClass ->
+        transform = {
+          it.safeAs<KtClass>()?.takeIf(::isHigherKindedType)?.identifyingElement
+        },
+        message = { identifier ->
           """
-          |${classOrInterface.name} is a Higher Kinded Type that may be used in polymorphic functions expressed over [Kind<F, A>] and with the type classes.
+          |${identifier.text} is a Higher Kinded Type that may be used in polymorphic functions expressed over [Kind<F, A>] and with the type classes.
           |For more info visit [https://arrow-kt.io/docs/patterns/glossary/#type-constructors](https://arrow-kt.io/docs/patterns/glossary/#type-constructors)
           |""".trimMargin()
         }
       ),
       addLineMarkerProvider(
         icon = ArrowIcons.POLY,
-        matchOn = { it.safeAs<KtNamedFunction>()?.takeIf(KtNamedFunction::isKindPolymorphic)?.takeIf(KtNamedFunction::hasExtensionDefaultValues) },
+        transform = { it.safeAs<KtNamedFunction>()?.takeIf(KtNamedFunction::isKindPolymorphic)?.takeIf(KtNamedFunction::hasExtensionDefaultValues) },
         message = { f: KtNamedFunction ->
           FuncScope(f).run {
             val target = ScopedList(receiver.value, transform = { it.text.escapeHTML() })
-            """|<h1>$name</h1>
-                   |<code lang="kotlin">
-                   |  ${this@run} 
-                   |</code>
-                   |<code>$name</code> is a polymorphic function that can be applied over $target.
-                   |<code>$name</code> requires @extension proofs for <code>${givenParameters()}</code>
-                   |
-                   |Valid <code>@extension</code> locations:
-                   |${validExtensionLocations()}
-                   |
-                   |<a href="https://arrow-kt.io/docs/patterns/polymorphic_programs/">Learn more about type class and ad-hoc polymorphism</a>
-                   |""".trimMargin()
+            val compose: List<Node> = h1(name) + kotlin(this@run) +
+              code(name) + text(" is a polymorphic function that can be applied over $target.") +
+              code(name) + text(" requires @extension proofs for ") + code(givenParameters()) + text("Valid ") +
+              code("@extension") + text("locations:\n${validExtensionLocations()}") +
+              a(href = "https://arrow-kt.io/docs/patterns/polymorphic_programs/") { "Learn more about type class and ad-hoc polymorphism" }
+            html(compose).render()
```

Comment:
@raulraja I added Html support. This renders like this for :
`compose.render()`
![Screenshot 2019-10-27 at 12 34 17](https://user-images.githubusercontent.com/46971368/67634174-c9833180-f8b8-11e9-999d-89010472e353.png)

or as in line `54`
![Screenshot 2019-10-27 at 12 36 48](https://user-images.githubusercontent.com/46971368/67634184-d6a02080-f8b8-11e9-8dbb-430baa22c255.png)

We should be able to add GoTo Actions to the Web, so you dont have to add the link at the end. And the User simply clicks on the Icon.

Reply:
Please specify, which style works for you rn :)

Reply:
Either style works and user should be able to manually generate html or use the jetbrains library equally. I for one prefer string and htnl over dsls but I think many would appreciate the HTML based DSL